src/Screens/Flows/App/NFT/NFTReportCollection/ReportNFTTransactionScreen.tsx

```diff
@@ -0,0 +1,161 @@
+import { useNavigation } from "@react-navigation/native"
+import { NativeStackScreenProps } from "@react-navigation/native-stack"
+import React, { useCallback } from "react"
+import { StyleSheet } from "react-native"
+import {
+    BaseSpacer,
+    BaseText,
+    BaseView,
+    DelegationView,
+    FadeoutButton,
+    GasFeeSpeed,
+    Layout,
+    RequireUserPassword,
+} from "~Components"
+import { AnalyticsEvent } from "~Constants"
+import { useAnalyticTracking, useBottomSheetModal, useTheme, useTransactionScreen } from "~Hooks"
+import { useI18nContext } from "~i18n"
+import { Routes } from "~Navigation"
+import { RootStackParamListNFT } from "~Navigation/Stacks/NFTStack"
+import { selectCollectionWithContractAddress, setIsAppLoading, useAppDispatch, useAppSelector } from "~Storage/Redux"
+import { AddressUtils } from "~Utils"
+import { NFTReportSuccessBottomsheet } from "./NFTReportSuccessBottomsheet"
+
+type Props = NativeStackScreenProps<RootStackParamListNFT, Routes.REPORT_NFT_TRANSACTION_SCREEN>
+
+export const ReportNFTTransactionScreen = ({ route }: Props) => {
+    const { LL } = useI18nContext()
+    const nav = useNavigation()
+    const track = useAnalyticTracking()
+    const dispatch = useAppDispatch()
+    const theme = useTheme()
+    const { ref: successBottomSheetRef, onOpen: openBottomSheet, onClose: closeBottomSheet } = useBottomSheetModal()
+
+    const { nftAddress, transactionClauses } = route.params
+
+    const collection = useAppSelector(state => selectCollectionWithContractAddress(state, nftAddress))
+
+    const handleBottomSheetClose = useCallback(() => {
+        closeBottomSheet()
+        nav.goBack()
+    }, [closeBottomSheet, nav])
+
+    const onTransactionSuccess = useCallback(() => {
+        dispatch(setIsAppLoading(false))
+        openBottomSheet()
+    }, [dispatch, openBottomSheet])
+
+    const onTransactionFailure = useCallback(() => {
+        dispatch(setIsAppLoading(false))
+        nav.goBack()
+    }, [dispatch, nav])
+
+    const {
+        selectedDelegationOption,
+        onSubmit,
+        isPasswordPromptOpen,
+        handleClosePasswordModal,
+        onPasswordSuccess,
+        setSelectedFeeOption,
+        selectedFeeOption,
+        resetDelegation,
+        setSelectedDelegationAccount,
+        setSelectedDelegationUrl,
+        selectedDelegationAccount,
+        selectedDelegationUrl,
+        isDisabledButtonState,
+        gasOptions,
+        gasUpdatedAt,
+        isGalactica,
+        isBaseFeeRampingUp,
+        speedChangeEnabled,
+    } = useTransactionScreen({
+        clauses: transactionClauses,
+        onTransactionSuccess,
+        onTransactionFailure,
+        initialRoute: Routes.NFTS,
+    })
+
+    const formattedAddress = AddressUtils.humanAddress(nftAddress, 10, 10)
+
+    const handleConfirm = useCallback(() => {
+        track(AnalyticsEvent.NFT_COLLECTION_REPORT_CONFIRMED, {
+            nftAddress: nftAddress,
+        })
```

Comment:
Should we move this to `onTransactionSuccess`? We could have false positives here